### PR TITLE
Use credential provider chain to allow credentials_process

### PR DESCRIPTION
**Pull request by wilsonjackson**
This uses the `CredentialProviderChain` to allow `ProcessCredentials` to be sourced from shared AWS config. This allows SAML-federated users that can only obtain temporary credentials to be used with parameter-store-manager.

### CODE REVIEW

File: app/clients/aws.js [hunk 1]
```diff
@@ -4,12 +4,18 @@ import localStore from '../store/localStore';
...
-const credentials = new AWS.SharedIniFileCredentials({
-  profile: localStore.get('profile')
+const profile = localStore.get('profile');
+const sharedIniFileCredentials = new AWS.SharedIniFileCredentials({ profile });
+const processCredentials = new AWS.ProcessCredentials({ profile });
+
+const chain = new AWS.CredentialProviderChain();
+chain.providers.push(sharedIniFileCredentials);
+chain.providers.push(processCredentials);
+chain.resolve((err, credentials) => {
```
Comment (wilsonjackson) [inline app/clients/aws.js]: I wasn't sure of the best way to handle/surface a potential error. Suggestions?

File: app/clients/aws.js [hunk 2]
```diff
@@ -4,12 +4,18 @@ import localStore from '../store/localStore';
 
 process.env.AWS_SDK_LOAD_CONFIG = true;
 
-const credentials = new AWS.SharedIniFileCredentials({
-  profile: localStore.get('profile')
+const profile = localStore.get('profile');
+const sharedIniFileCredentials = new AWS.SharedIniFileCredentials({ profile });
+const processCredentials = new AWS.ProcessCredentials({ profile });
```
Comment (wilsonjackson) [inline app/clients/aws.js]: For now I've only elected to add `ProcessCredentials` as that's my required use case, but other credential providers could be added to this chain to fully replicate the behavior of the AWS CLI. I figured there would be a default credential provider chain in the AWS SDK that I could use, but I didn't see one...

**Comment by smblee** [comment]
Apologies; just got a chance to take a look at this.

File: app/clients/aws.js (hunk 2 above)
Comment (smblee) [reply app/clients/aws.js]: No worries; I think this makes sense. Let's add more providers as we uncover more usecases!

**Comment by smblee** [comment]
Very cool; was not aware you could chain up different credential providers. I will check the backwards compatibility with my current use cases (local config file/sts key) and if that looks good, let's merge it in.

**Comment by smblee** [comment]
The code was not supporting the old use case so i made some changes to support that. Please make sure that my changes still works with your use case.

Also with this change, the credentials are also loaded every time (except when the credentials are valid because AWS client smartly caches them), so if credentials fail to load because e.g. non-existent profile in the setting, user should be able to set them up properly and refresh to use the new setting without having to restart the program.

neat!

**Comment by smblee** [comment]
Made a ticket to improve AWS credential errors so we can add more potential cred providers as needed.

https://github.com/smblee/parameter-store-manager/issues/103

**Pull request merged**